### PR TITLE
Add American time format to lua getDateTime()

### LINKED ISSUE #5430: Would it be possible to add a 12 hour clock in firmware or lua?
Would it be possible to add a 12 hour clock in firmware or lua?

### PR DESCRIPTION
**Pull request by 3djc**
Conversion rules as per : https://www.timeanddate.com/time/am-and-pm.html

Will adjust lua doc when merged

Fixes #5430

### CODE REVIEW

**Comment by opentx-sonar** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [api_general.cpp#L786](https://github.com/opentx/opentx/blob/c296a4e77890061006bf6412be20107d6c267dfd/radio/src/lua/api_general.cpp#L786): Remove the unused function parameter "L". [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1172)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: radio/src/lua/api_general.cpp [hunk 1]
```diff
@@ -122,13 +122,28 @@ static int luaGetTime(lua_State * L)
 static void luaPushDateTime(lua_State * L, uint32_t year, uint32_t mon, uint32_t day,
                             uint32_t hour, uint32_t min, uint32_t sec)
 {
-  lua_createtable(L, 0, 6);
+  uint32_t hour12 = hour;
+
+  if (hour < 1) {
+    hour12 = hour;
```
Comment (mpaperno) [inline radio/src/lua/api_general.cpp]: I think you meant `hour12 = 12`?

Comment (3djc) [reply radio/src/lua/api_general.cpp]: Indeed !

**Comment by opentx-sonar** [comment]
SonarQube analysis reported 14 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [api_general.cpp#L218](https://github.com/opentx/opentx/blob/86469ad45d3051c8672e8f6a3e2434f5cef7e991/radio/src/lua/api_general.cpp#L218): End this switch case with an unconditional break, continue, return or throw statement. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ANonEmptyCaseWithoutBreak)
1. ![CRITICAL][CRITICAL] [api_general.cpp#L248](https://github.com/opentx/opentx/blob/86469ad45d3051c8672e8f6a3e2434f5cef7e991/radio/src/lua/api_general.cpp#L248): Refactor this function to reduce its Cognitive Complexity from 39 to the 25 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3776)
1. ![MAJOR][MAJOR] [api_general.cpp#L172](https://github.com/opentx/opentx/blob/86469ad45d3051c8672e8f6a3e2434f5cef7e991/radio/src/lua/api_general.cpp#L172): Remove the unused function parameter "telemetrySensor". [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1172)
1. ![MAJOR][MAJOR] [api_general.cpp#L182](https://github.com/opentx/opentx/blob/86469ad45d3051c8672e8f6a3e2434f5cef7e991/radio/src/lua/api_general.cpp#L182): Remove the unused function parameter "telemetrySensor". [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1172)
1. ![MAJOR][MAJOR] [api_general.cpp#L188](https://github.com/opentx/opentx/blob/86469ad45d3051c8672e8f6a3e2434f5cef7e991/radio/src/lua/api_general.cpp#L188): Remove the unused function parameter "telemetrySensor". [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1172)
1. ![MAJOR][MAJOR] [api_general.cpp#L1029](https://github.com/opentx/opentx/blob/86469ad45d3051c8672e8f6a3e2434f5cef7e991/radio/src/lua/api_general.cpp#L1029): Refactor this method to not always return the same value. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3516)
1. ![MAJOR][MAJOR] [api_general.cpp#L1079](https://github.com/opentx/opentx/blob/86469ad45d3051c8672e8f6a3e2434f5cef7e991/radio/src/lua/api_general.cpp#L1079): Refactor this method to not always return the same value. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3516)
1. ![MAJOR][MAJOR] [api_general.cpp#L1158](https://github.com/opentx/opentx/blob/86469ad45d3051c8672e8f6a3e2434f5cef7e991/radio/src/lua/api_general.cpp#L1158): Remove this block of commented out code. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ACommentedCode)
1. ![MINOR][MINOR] [api_general.cpp#L196](https://github.com/opentx/opentx/blob/86469ad45d3051c8672e8f6a3e2434f5cef7e991/radio/src/lua/api_general.cpp#L196): Uppercase the literal suffix. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ALiteralSuffix)
1. ![MINOR][MINOR] [api_general.cpp#L238](https://github.com/opentx/opentx/blob/86469ad45d3051c8672e8f6a3e2434f5cef7e991/radio/src/lua/api_general.cpp#L238): Uppercase the literal suffix. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ALiteralSuffix)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by opentx-sonar** [comment]
SonarQube analysis reported 14 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [api_general.cpp#L218](https://github.com/opentx/opentx/blob/8ab48a9af3ddf53ea4448442b3bf449e3f71cf45/radio/src/lua/api_general.cpp#L218): End this switch case with an unconditional break, continue, return or throw statement. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ANonEmptyCaseWithoutBreak)
1. ![CRITICAL][CRITICAL] [api_general.cpp#L248](https://github.com/opentx/opentx/blob/8ab48a9af3ddf53ea4448442b3bf449e3f71cf45/radio/src/lua/api_general.cpp#L248): Refactor this function to reduce its Cognitive Complexity from 39 to the 25 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3776)
1. ![MAJOR][MAJOR] [api_general.cpp#L172](https://github.com/opentx/opentx/blob/8ab48a9af3ddf53ea4448442b3bf449e3f71cf45/radio/src/lua/api_general.cpp#L172): Remove the unused function parameter "telemetrySensor". [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1172)
1. ![MAJOR][MAJOR] [api_general.cpp#L182](https://github.com/opentx/opentx/blob/8ab48a9af3ddf53ea4448442b3bf449e3f71cf45/radio/src/lua/api_general.cpp#L182): Remove the unused function parameter "telemetrySensor". [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1172)
1. ![MAJOR][MAJOR] [api_general.cpp#L188](https://github.com/opentx/opentx/blob/8ab48a9af3ddf53ea4448442b3bf449e3f71cf45/radio/src/lua/api_general.cpp#L188): Remove the unused function parameter "telemetrySensor". [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1172)
1. ![MAJOR][MAJOR] [api_general.cpp#L1029](https://github.com/opentx/opentx/blob/8ab48a9af3ddf53ea4448442b3bf449e3f71cf45/radio/src/lua/api_general.cpp#L1029): Refactor this method to not always return the same value. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3516)
1. ![MAJOR][MAJOR] [api_general.cpp#L1079](https://github.com/opentx/opentx/blob/8ab48a9af3ddf53ea4448442b3bf449e3f71cf45/radio/src/lua/api_general.cpp#L1079): Refactor this method to not always return the same value. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3516)
1. ![MAJOR][MAJOR] [api_general.cpp#L1158](https://github.com/opentx/opentx/blob/8ab48a9af3ddf53ea4448442b3bf449e3f71cf45/radio/src/lua/api_general.cpp#L1158): Remove this block of commented out code. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ACommentedCode)
1. ![MINOR][MINOR] [api_general.cpp#L196](https://github.com/opentx/opentx/blob/8ab48a9af3ddf53ea4448442b3bf449e3f71cf45/radio/src/lua/api_general.cpp#L196): Uppercase the literal suffix. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ALiteralSuffix)
1. ![MINOR][MINOR] [api_general.cpp#L238](https://github.com/opentx/opentx/blob/8ab48a9af3ddf53ea4448442b3bf449e3f71cf45/radio/src/lua/api_general.cpp#L238): Uppercase the literal suffix. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ALiteralSuffix)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: radio/src/lua/api_general.cpp [hunk 2]
```diff
@@ -122,13 +122,28 @@ static int luaGetTime(lua_State * L)
...
+  }
+  lua_createtable(L, 0, 8);
   lua_pushtableinteger(L, "year", year);
   lua_pushtableinteger(L, "mon", mon);
   lua_pushtableinteger(L, "day", day);
   lua_pushtableinteger(L, "hour", hour);
   lua_pushtableinteger(L, "min", min);
   lua_pushtableinteger(L, "sec", sec);
+  lua_pushtableinteger(L, "hour12", hour12);
+  if(hour < 12) {
```
Comment (bsongis) [inline radio/src/lua/api_general.cpp]: a space missing here

**Review by bsongis — APPROVED** [review]

**Pull request merged**